capa/features/extractors/base_extractor.py

```diff
@@ -100,6 +123,13 @@ def get_base_address(self) -> Union[AbsoluteVirtualAddress, capa.features.addres
         """
         raise NotImplementedError()
 
+    @abc.abstractmethod
+    def get_sample_hashes(self) -> SampleHashes:
+        """
+        fetch the hashes for the sample contained within the extractor.
+        """
+        raise NotImplementedError()
```

Comment:
I think for dynamic extractors the hashes are almost always going to be in the report, so it'd be better to internalize that hash extraction rather than opening the file before passing to the extractor, getting the hashes, and passing all of that back to the extractor which is going to reopen the report anyways...

As for static extractors, I think it'd be nice if we make them similar to the dynamic extractors, and keep the hashes' extraction internal to the extractor.

Reply:
yeah that looks pretty awesome! :)

I'll add it.